src/protocols/ranking.js

```diff
@@ -70,16 +71,11 @@ class Ranking {
 			return;
 		}
 
-		let nexVersion;
-		if (packet.connection.title.nex_ranking_version) {
-			nexVersion = packet.connection.title.nex_ranking_version;
-		} else {
-			nexVersion = packet.connection.title.nex_version;
-		}
+		const nexVersion = packet.connection.title.nex_ranking_version || packet.connection.title.nex_version;
 
 		// Check if game uses legacy Ranking. Since there aren't many games that use it,
 		// assume games not listed on title list to use modern Ranking
-		if (nexVersion.major < 3 && nexVersion.major > 0) {
+		if (semver.lt(nexVersion, '3.0.0')) {
```

Comment:
Here we should check if the version is greater than 0.0.0 to use modern Ranking by default

Reply:
This PR also includes changes which make every titles protocol library version match the games NEX version, for now. Since this is the most common case for games, with games having differing versions being the vast minority, it felt easier to make everything match and then make patch the games which have different versions as we find them

Because of that, there are only 4 games which have everything set `0.0.0`, and only because we don't have their NEX data listed in the files used by the json generator. Thos games being

- Kid Icarus Uprising
- Luigi's Mansion 2
- Pokémon X/Y
- Tri Force Heroes

So if we only check for being above `0.0.0` then all games, even those which don't use the modern versions (like Mario Kart 7, whose version is now set to `2.4.3`) would not use the old protocols

Reply:
I think there is a misunderstanding. I meant that you keep the check for below 3.0 *and* we also add a check for versions greater than 0 (which is the default for not listed games). `semver.lt(nexVersion, '3.0.0') && semver.gt(nexVersion, '0.0.0')`.

With this we assume by default that every not listed game uses the new Ranking protocol, which would probably be the majority of games that don't have their NEX versions listed.

Reply:
You're right I misunderstood, I'm on the same page now!